### ISSUE TITLE
Overlay Webs to Better Compare Players

### ISSUE THREAD

**Issue by nylon22** [marcusdarmstrong/mockdraftable-web#1]
I think it would be helpful to compare two players if you could overlay the web of one player over another in a different color. 

Question:
* How to render webs if player one has results in a measurable player two does not

If this is an idea you're interested in, lmk and I'll submit a PR.

**Comment by marcusdarmstrong** [comment]
Hey there! 

So yeah, this is a feature that's definitely on my backlog (it's very highly requested). It needs some thought as to how to integrate it into the site from a UX perspective, in addition to the question you raised. So if you're open to some discussion on that, I'd very much welcome a PR.

**Comment by nylon22** [comment]
Awesome! Yeah, I figure UX-wise, it seems pretty straightforward to render a "Compare" button next to each of the player comparisons. Beyond that, I definitely agree searching for players to compare to needs some thought. One question I'd have is if the search should filter players whose `playerPosition` doesn't match the `selectedPosition` of player 1. Also, of course, where it would make sense to render that search bar

**Comment by marcusdarmstrong** [comment]
Yeah, so I'm not big on per-player comparison buttons, primary because of space constraints on mobile (which the current layout is already running out of in rare cases)
<img width="338" alt="screen shot 2017-03-08 at 1 49 47 pm" src="https://cloud.githubusercontent.com/assets/3826925/23723373/26c13c48-0406-11e7-8628-e91f1fddd6d0.png">

But yeah, in theory a head-to-head comparison page needs *two* (or more?) selected players and a selected position ID. 

Should be able to re-use the position selector pretty easily, and default it to ATH. As long as there's something in the UI to display the two players' "primary" positions it should be good.

Currently the player typeahead is semi-global, in that you couldn't really have two of them on the same page, since the results/etc is global. That's a solvable thing but yeah (And now that I'm thinking about it, it probably should be moved out of the redux store to some extent, but whatever), it still probably makes sense to only bother with one typeahead on an overlay page.

We'd also need some way of distinguishing between the two graphs on the overlay. Color is primarily already taken to describe position throughout the site, so I'm not keen to overload there. So that really leaves two obvious options... Line style (dots/dashes/etc) and the containers for the percentiles... Could go Circle -> Square -> Pentagon -> Hexagon -> etc. I'm partial to the later, since it'll be less visually jarring to see un-closed polygons for the spidergraphs, and it's fairly easy to convey the Player -> Shape mapping.

Seems like a good place to start here is just getting a working comparisons page up, and then we can figure out how to link to it in the application afterwards—I'm currently messing around with a position aggregate data page, and that also is lacking tie-in to the rest of the app, so it may be that I need to add a hamburger icon somewhere.

**Comment by nylon22** [comment]
Wasn't thinking about mobile for per-player comparison buttons - good point. I do agree with using the containers for the percentiles direction to distinguish players. But yeah, I'll start hacking on a player comparison page and we can worry about integrating with the rest of the site later. Thanks for the input!

**Comment by marcusdarmstrong** [comment]
Sure thing.

You'll need a Postgres DB set up to actually run the site locally with a connection string in the DATABASE_URL env variable (or .env file in the repo). The schema.sql in the repo is valid, but I'll take some time tonight to extract out a little test data for you.

**Comment by marcusdarmstrong** [comment]
Oh, and idk how much you've gotten into the code just yet but it's a hand-rolled router system that's still got some warts, but the basic idea is that the current url is a function of the redux state (and vis versa), so `router.js` translates the current/request url into a series of redux actions to build the initial page state (so that's how you'd go about adding a new page to the app).

**Comment by nylon22** [comment]
Interesting... have you thought about using react router? https://github.com/ReactTraining/react-router. It works really nicely for me. And I'll have to do some homework on redux to get up to speed. I've only used mobx for state management with react. Thanks for setting me up with some test data!

**Comment by marcusdarmstrong** [comment]
Yeah, I looked into it in the early stages and decided against it—Server side rendering is second class citizen of the project, and it's got some pretty opinionated takes on stuff that I disagree with (Hash routing as a fallback, props for route parents, etc). 

I've added ./test-data.dump to the repo, you can follow the instructions here: https://devcenter.heroku.com/articles/heroku-postgres-import-export#restore-to-local-database for loading it locally. There might be some extra tables in there, not really sure, so just lemme know if it doesn't go well.

**Comment by nylon22** [comment]
Hey. I got postgres setup with the test-data:

`mockdraftable=# \dt`

                 List of relations
 Schema |          Name          | Type  |  Owner
--------|------------------------|-------|------
 public | play_evolutions        | table | twarner
 public | t_measurement          | table | twarner
 public | t_member               | table | twarner
 public | t_player               | table | twarner
 public | t_position_eligibility | table | twarner
 public | testings               | table | twarner
(6 rows)

But when I run `npm start`, I get:

```
[WARN] No ENV file found
11:05:40 PM web.1 |  Starting Player Load
11:05:40 PM web.1 |  error: database "User ID=twarner;Host=localhost;Port=5432;Database=mockdraftable" does not exist
...
```

Not sure why I'm getting this because the dbname, port, and userid look correct to me. I haven't used postgres very often, so I'm not sure if it's because I don't have an ENV file, which looks like it's in `.gitignore`. However, I did set my `DATABASE_URL` env variable to `User ID=twarner;Host=localhost;Port=5432;Database=mockdraftable;`

Any thoughts?

**Comment by marcusdarmstrong** [comment]
So the whole ENV file thing is just a way to provide a file that lists out environment variables in the top level of the project that get injected prior to it starting up; it doesn't actually matter if you're specifying them in your actual env.

Is that string in there your DATABASE_URL? The format mine uses is: "postgres://user:pass@host:5432/dbname"

**Comment by nylon22** [comment]
Hey! So I have the site running locally now. I had to do a couple things to get it running:

I had to add the table `t_school` that was missing in `test-data.dump` using this:

https://github.com/marcusdarmstrong/mockdraftable-web/blob/master/src/schema.sql#L12-L15

And I had to add `school_id` to `t_player` that was missing in `test-data.dump`:

https://github.com/marcusdarmstrong/mockdraftable-web/blob/master/src/schema.sql#L9

**Comment by marcusdarmstrong** [comment]
Awesome! Sorry about that; I should make a new data dump on the full/actual DB now that I've fixed the startup time (idk if you've pulled in the latest changes, but I sped it up by ~50x).
